src/Microsoft.ServiceFabric.Actors/Runtime/ActorStateProviderHelper.cs

```diff
@@ -224,15 +224,35 @@ internal static IActorStateProvider CreateDefaultStateProvider(ActorTypeInformat
             IActorStateProvider stateProvider = new NullActorStateProvider();
             if (actorTypeInfo.StatePersistence.Equals(StatePersistence.Persisted))
             {
+                bool isMigrationTarget = Utility.IsMigrationTarget(new List<Type>() { actorTypeInfo.ImplementationType });
+                bool isMigrationSource = Utility.IsMigrationSource(new List<Type>() { actorTypeInfo.ImplementationType });
 #if DotNetCoreClr
-                // Use KVSActorStatePRovide for Windows.
                 if (System.Runtime.InteropServices.RuntimeInformation.IsOSPlatform(System.Runtime.InteropServices.OSPlatform.Windows))
                 {
+                    if (isMigrationTarget)
+                    {
+                        var message = "Migration target attribute is valid only for Reliable Collection (RC) service. This is a KVS service";
+                        ActorTrace.Source.WriteWarning("ActorStateProviderHelper", message);
+                    }
+
                     stateProvider = new KvsActorStateProvider();
                 }
                 else
                 {
-                   stateProvider = new ReliableCollectionsActorStateProvider();
+                    if (isMigrationTarget)
+                    {
+                        stateProvider = new MigrationActorStateProvider();
+                    }
+                    else if (isMigrationSource)
+                    {
+                        var message = "Migration source attribute is valid only for KVS service. This is an RC service ";
+                        ActorTrace.Source.WriteWarning("ActorStateProviderHelper", message);
+                        stateProvider = new ReliableCollectionsActorStateProvider();
+                    }
+                    else
+                    {
+                        stateProvider = new ReliableCollectionsActorStateProvider();
+                    }
                 }
 #else
                 stateProvider = new KvsActorStateProvider();

```

Comment:
Don't you need to check MigrationSource and MigrationTarget for full framework as well ?